### DISCUSSION TITLE
posts/go-private-package/

### DISCUSSION THREAD

**Discussion by giscus** [yunnysunny/yunnysunny.github.io#24] [Show and tell]
# posts/go-private-package/

使用私有托管的 go 包仓库的折腾之路

https://blog.whyun.com/posts/go-private-package/

**Comment by lilynnli** [comment]
Hi！感谢提示。 But ```.netrc 文件``` 相关配置中，在windows环境下，文件名应为```_netrc``` :smile:

**Reply by yunnysunny** [reply]
谢谢反馈

**Comment by dbing** [comment]
写的很棒 👍🏻，请问你的日常开发流程是这样吗？1. 开发阶段是在 go.mod 文件中使用 replace，2.在私有包增加代码，3. 提交代码后先给私有包打tag，go get 后再push主库代码

**Reply by yunnysunny** [reply]
replace语句只有在特殊情况下使用。 就是gitlab运行在容器中 容器外侧有一个反向代理服务做反代 并且容器开启的端口和反向代理端口不一致，才有可能使用replace语句。具体参见你的报错信息。

**Reply by yunnysunny** [reply]
在被引用项目中打tag 然后在引用项目中安装tag版本的被引用项目 这个流程是固定的 即使你的项目在github上 也是这个流程

**Reply by dbing** [reply]
感谢作者解答，请教一下，开发阶段，私有包在本地，开发时项目需要用到私有包中的代码，IDE中为了可以跳转到私有包，开发时本地要配置 replace，开发结束后，提交代码到git版本库前 go.mod 文件中的 replace 又得移除掉，本地开发时又得增加上 replace，感觉有些麻烦，请问有没有更好的办法处理呢？

**Reply by yunnysunny** [reply]
我都是将代码先提交到 git 仓库中，然后再在引用项目中引用 git 仓库中提交的版本。

**Reply by dbing** [reply]
感谢解答，再在引用项目中引用，是这么这样一个流程吗？
1. 先提交私有库代码到git仓库；
2. 引用的项目中 go get package 引用最新提交的代码
3. 提交项目库代码。

**Reply by yunnysunny** [reply]
B项目 中引入了 A 库，如果 B 期望 A 库增加新功能，则先写 A 库的代码，然后把 A 库的代码打新 tag 并 push 到 git 仓库。
在 B 项目中安装 A 库的新 tag 版本。
提交 B 项目的代码。

**Reply by dbing** [reply]
感谢解答👍
